tslib/src/functions/check-sync/prepared-check-sync.data.service.ts

```diff
@@ -0,0 +1,120 @@
+import { SqlService } from '../../sql/sql.service'
+import { TYPES } from 'mssql'
+
+export interface IPreparedCheckSyncDataService {
+  getActiveCheckReferencesByPupilUuid (pupilUUID: string): Promise<IActiveCheckReference[]>
+  getAccessArrangementsCodesByIds (aaIds: Array<number>): Promise<Array<string>>
+  getAccessArrangementsByCheckCode (checkCode: string): Promise<Array<any>>
+}
+
+export class PreparedCheckSyncDataService implements IPreparedCheckSyncDataService {
+  private static aaCodes = new Array<IAccessArrangementCode>()
+  private sqlService: SqlService
+
+  constructor () {
+    this.sqlService = new SqlService()
+  }
+
+  async getActiveCheckReferencesByPupilUuid (pupilUUID: string): Promise<IActiveCheckReference[]> {
+    const liveCheckSql = `
+    SELECT TOP 1 chk.checkCode, pn.val as pupilPin, s.pin as schoolPin
+      FROM [mtc_admin].[check] chk
+    INNER JOIN [mtc_admin].checkStatus cs
+      ON cs.id = chk.checkStatus_id
+    INNER JOIN mtc_admin.pupil p
+      ON p.id = chk.pupil_id
+    INNER JOIN [mtc_admin].school s
+      ON s.id = p.school_id
+    INNER JOIN [mtc_admin].checkPin cp
+      ON chk.id = cp.check_id
+    INNER JOIN [mtc_admin].pin pn
+      ON cp.pin_id = pn.id
+    WHERE chk.isLiveCheck = 1
+      AND cs.code NOT IN ('CMP', 'EXP', 'NTR')
+      AND p.urlSlug='${pupilUUID}'
```

Comment:
This isn't a param despite it being passed in from outside the function. Why not?